Mathlib/LinearAlgebra/LinearIndependent/Basic.lean

```diff
@@ -513,89 +513,65 @@ theorem linearIndependent_monoidHom (G : Type*) [MulOneClass G] (L : Type*) [Com
-                _ =
-                    (g a * a x * a y + ∑ i ∈ s, g i * i x * i y) -
-                      (g a * a x * a y + ∑ i ∈ s, g i * a x * i y) := by
-                  rw [add_sub_add_left_eq_sub]
-                _ =
-                    (∑ i ∈ insert a s, g i * i x * i y) -
-                      ∑ i ∈ insert a s, g i * a x * i y := by
-                  rw [Finset.sum_insert has, Finset.sum_insert has]
-                _ =
-                    (∑ i ∈ insert a s, g i * i (x * y)) -
-                      ∑ i ∈ insert a s, a x * (g i * i y) := by
-                  congrm ∑ i ∈ insert a s, ?_ - ∑ i ∈ insert a s, ?_
-                  · rw [map_mul, mul_assoc]
-                  · rw [mul_assoc, mul_left_comm]
-                _ =
-                    (∑ i ∈ insert a s, (g i • (i : G → L))) (x * y) -
-                      a x * (∑ i ∈ insert a s, (g i • (i : G → L))) y := by
-                  rw [Finset.sum_apply, Finset.sum_apply, Finset.mul_sum]; rfl
-                _ = 0 - a x * 0 := by rw [hg]; rfl
-                _ = 0 := by rw [mul_zero, sub_zero]
-                )
-              i his
-        -- On the other hand, since `a` is not already in `s`, for any character `i ∈ s`
-        -- there is some element of the monoid on which it differs from `a`.
-        have h2 : ∀ i : G →* L, i ∈ s → ∃ y, i y ≠ a y := fun i his =>
-          Classical.by_contradiction fun h =>
-            have hia : i = a := MonoidHom.ext fun y =>
-              Classical.by_contradiction fun hy => h ⟨y, hy⟩
-            has <| hia ▸ his
-        -- From these two facts we deduce that `g` actually vanishes on `s`,
-        have h3 : ∀ i ∈ s, g i = 0 := fun i his =>
-          let ⟨y, hy⟩ := h2 i his
-          have h : g i • i y = g i • a y := congr_fun (h1 i his) y
-          Or.resolve_right (mul_eq_zero.1 <| by rw [mul_sub, sub_eq_zero]; exact h)
-            (sub_ne_zero_of_ne hy)
-        -- And so, using the fact that the linear combination over `s` and over `insert a s` both
-        -- vanish, we deduce that `g a = 0`.
-        have h4 : g a = 0 :=
-          calc
-            g a = g a * 1 := (mul_one _).symm
-            _ = (g a • (a : G → L)) 1 := by rw [← a.map_one]; rfl
-            _ = (∑ i ∈ insert a s, (g i • (i : G → L))) 1 := by
-              rw [Finset.sum_eq_single a]
-              · intro i his hia
-                rw [Finset.mem_insert] at his
-                rw [h3 i (his.resolve_left hia), zero_smul]
-              · intro haas
-                exfalso
-                apply haas
-                exact Finset.mem_insert_self a s
-            _ = 0 := by rw [hg]; rfl
-        -- Now we're done; the last two facts together imply that `g` vanishes on every element
-        -- of `insert a s`.
-        (Finset.forall_mem_insert ..).2 ⟨h4, h3⟩
+  apply linearIndependent_iff'.2
+  intro s
+  induction s using Finset.induction_on with
+  | empty => simp
+  | insert has ih =>
+  rename_i a s
```

Comment:
```suggestion
  | @insert a s has ih =>
```
works

Reply:
`insert {a s} has ih` might work too (although I haven't tested it here)